sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ShowCreateTableExec.scala

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources.v2
+
+import scala.collection.mutable
+
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.catalog.BucketSpec
+import org.apache.spark.sql.catalyst.expressions.Attribute
+import org.apache.spark.sql.catalyst.util.escapeSingleQuotedString
+import org.apache.spark.sql.connector.catalog.{CatalogV2Util, Table, TableCatalog}
+import org.apache.spark.sql.connector.expressions.{BucketTransform, FieldReference, IdentityTransform}
+import org.apache.spark.sql.errors.QueryExecutionErrors
+import org.apache.spark.sql.execution.LeafExecNode
+import org.apache.spark.sql.execution.command.DDLUtils
+import org.apache.spark.unsafe.types.UTF8String
+
+/**
+ * Physical plan node for show create table.
+ */
+case class ShowCreateTableExec(
+    output: Seq[Attribute],
+    table: Table,
+    asSerde: Boolean = false) extends V2CommandExec with LeafExecNode {
+  override protected def run(): Seq[InternalRow] = {
+    val builder = StringBuilder.newBuilder
+    if (isView(table)) {
+      // It is better to use `ShowCreateViewExec` independent from `ShowCreateTableExec`
+      throw new AnalysisException("SHOW CREATE TABLE is not supported for V2 Views.")
```

Comment:
we can fail this in analyze phase, not here

Reply:
@yaooqinn ，If the view is temp view, it will throw AnalysisException. Becauce when  call visitShowCreateTable, temp view is not allowed.  

`override def visitShowCreateTable(ctx: ShowCreateTableContext): LogicalPlan = withOrigin(ctx) {
    ShowCreateTable(
      UnresolvedTableOrView(
        visitMultipartIdentifier(ctx.multipartIdentifier()),
        allowTempView = false),
      ctx.SERDE != null)
  }`

I'm not sure how to tell the view is permanent view ?